packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImageResizer.java

```diff
@@ -85,7 +85,9 @@ private File resizedImage(String path, Double maxWidth, Double maxHeight) throws
 
     Bitmap scaledBmp = Bitmap.createScaledBitmap(bmp, width.intValue(), height.intValue(), false);
     ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-    scaledBmp.compress(Bitmap.CompressFormat.JPEG, 100, outputStream);
+    boolean isPNG = path.toLowerCase().endsWith("png");
```

Comment:
This covers almost everything, but I think there may still be some edge cases where the image may have transparent pixels that are lost without the filename necessarily ending in `png`. What do you think about instead casing off of if the bitmap supports transparency? There's [`Bitmap#hasAlpha()`](https://developer.android.com/reference/android/graphics/Bitmap.html#hasAlpha()).

Reply:
That is a great suggestion. Thanks for reviewing and proposing a better alternative. Do you know if there is something similar we can use on iOS?

Reply:
For iOS I see there is this: https://developer.apple.com/documentation/coregraphics/1455401-cgimagegetalphainfo?language=objc - I will update the PR. Thanks again for jumping in 💯